tools/genmake2

```diff
@@ -1084,6 +1086,134 @@ EOF
     echo " --> set HAVE_NETCDF='$HAVE_NETCDF'" >> $LOGFILE
 }
 
+check_nf90io_mpi_libs()  {
+    if test ! "x$SKIP_NF90IO_CHECK" = x ; then
+	    return
+    fi
+    echo >> $LOGFILE
+    echo "running: check_nf90io_libs()" >> $LOGFILE
+    echo "First looking for hdf5 in nc-config:" >> $LOGFILE
+    echo "running: $NETCDF/bin/nc-config --has-hdf5" >> $LOGFILE
+    if [[ $($NETCDF/bin/nc-config --has-hdf5) != "yes" ]]; then
+      echo "nc-config could not fine hdf5" >> $LOGFILE
+      return
+    fi
+    echo "         ...returns yes" >> $LOGFILE
+
+    echo "Looking for h5pcc:" >> $LOGFILE
+    echo "running: $HDF5/bin/h5pcc -showconfig | grep Parallel" >> $LOGFILE
+    if [[ $($HDF5/bin/h5pcc -showconfig | grep Parallel) != *yes* ]]; then
+      echo "Somehow h5pcc not compiled w/ parallel" >> $LOGFILE
+      return
+    fi
+    echo "         ...returns yes" >> $LOGFILE
+
+    # now we check if the parallel tests can actually compile...
+    echo "===  f90tst_parallel.f90  >>>" > f90tst_parallel.log
+    #cat genmake_tnc.F >> f90tst_parallel.log
+    echo "<<<  f90tst_parallel.f90 ===" >> f90tst_parallel.log
+    echo "$FC $FFLAGS $FOPTIM -c f90tst_parallel.f90 \ " >> f90tst_parallel.log
+    echo "  &&  $LINK $FFLAGS $FOPTIM -o f90tst_parallel.o $LIBS" >> f90tst_parallel.log
+    $FC $FFLAGS $FOPTIM  $INCLUDES -c ${TOOLSDIR}/maketests/f90tst_parallel.f90 >> f90tst_parallel.log 2>&1  \
```

Comment:
Please change to F90C here and below.  Many optfiles assume that FC only sees fixed-form files and will not compile this.

Reply:
I think we effectively use FC for fixed-form source files (with file extension .F) and F90C for free-form ones (with extension .F90).  There can be Fortran-90 features in .F files and these may then just not work with some compilers (like g77).  @jm-c correct me if I'm wrong.